scripts/run_mypy.sh

```
@@ -21,20 +21,70 @@
+# function or needs a device; only symbol resolution does. The CUDA toolkit
+# ships a stub libcuda.so for exactly this case, so point the loader at it when
+# no real driver is present. Verified on a driverless node: with the stub the
+# bindings module imports and initializes cleanly, and torch correctly reports
+# zero devices.
+ensure_cuda_driver_stub() {
+    if python3 -c "import ctypes; ctypes.CDLL('libcuda.so.1')" 2>/dev/null; then
+        return  # A real driver is installed; leave the loader path alone.
+    fi
+    local stub
+    stub=$(ls /usr/local/cuda*/targets/*/lib/stubs/libcuda.so \
+              /usr/local/cuda*/lib64/stubs/libcuda.so 2>/dev/null | head -n 1)
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)